SYCL Resource

homerdin opened the pull request:
Initial PR to add a SYCL resource.

trws commented on include/camp/defines.hpp:
```diff
@@ -88,6 +88,13 @@ namespace camp
+#define CAMP_HAVE_SYCL 1
+#define CAMP_DEVICE
+#define CAMP_HOST_DEVICE
+#pragma warn("Sycl compiler!")
```
Do you mean to leave this in?

homerdin replied:
No, I'd forgotten to remove it.  I'll remove it now.

trws reviewed (commented):
This looks nice and clean, everything seems to be set up the way it should be too.  Just want to know if you intend that warning to stay, I'd probably say remove it, otherwise LGTM.

homerdin commented:
@trws I've updated the resource to work through/return a pointer to the sycl::queue.  Returning copies can cause issues with the memory associated with the queue.  

I've also updated the variable names, "stream" -> "qu"

trws commented:
Ok, this looks pretty good.  I need to figure out what's going on with appveyor though, almost looks like a cmake issue in the newer version over there.

trws commented:
Found the appveyor issue.  I'm going to merge this based on the results for linux, since the appveyor results are unrelated.  Thanks!
